frontend/app/src/types/dynamic-messages/index.ts

```diff
@@ -32,9 +32,11 @@ export type WelcomeSchema = z.infer<typeof WelcomeSchema>;
 
 const DashboardMessage = z.object({
   action: Action.optional(),
+  freeUserOnly: z.boolean().optional(),
   message: z.string(),
   messageHighlight: z.string().optional(),
   period: VisibilityPeriod,
+  premiumUserOnly: z.boolean().optional(),
```

Comment:
wouldn't it be better to have a target: 'free' | 'premium' | undefined
instead of two boolean values?

So that the undefined is all and the others are specific?